ERCS/erc-7558.md

```diff
@@ -0,0 +1,69 @@
+---
+eip: 7558
+title: Minimal Lockable Range NFTs
+description: A minimal interface for soulbinding multiple ERC-721 NFTs
+author: Hans Bergren (@0xCLARITY)
+discussions-to: https://ethereum-magicians.org/t/erc-7558-minimal-lockable-range-nfts/16668
+status: Draft
+type: Standards Track
+category: ERC
+created: 2023-11-15
+requires: 165, 721
+---
+
+## Abstract
+
+This proposal extends [ERC-721](./eip-721.md). It proposes a minimal interface to make both individual tokens and a contiguous range of tokenIds soulbound (locked).
+
+## Motivation
+
+Other existing soulbound / lockable standards emit events that only include a single `tokenId`. This makes it gas-prohibitive to lock or unlock all of the tokens in any contiguous range, including potentially locking/unlocking the entire 721 contract.
+
+This proposal captures the use-cases covered by existing proposals, yet also allows for range operations around soulbinding/locking tokens.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+### Contract Interface
+
+The interface is defined as follows:
+
+```solidity
+// ERC165 interfaceId 0x75587558
+interface IERC7558 is IERC5192 {
+  /// @notice Emitted when the locking status for a range of tokens is changed to locked.
+  event RangeLocked(uint256 _fromTokenId, uint256 _toTokenId);
+
+  /// @notice Emitted when the locking status for a range of tokens is changed to unlocked.
+  event RangeUnlocked(uint256 _fromTokenId, uint256 _toTokenId);
+}
+```
+
+The Locked/Unlocked or RangeLocked/RangeUnlocked event MUST be emitted when the locking status of a token, or a consecutive range of tokens, is changed.
```

Comment:
Please explicitly state when each type of event must be emitted. Do `Locked` / `Unlocked` events need to be emitted for every token in the range? I would assume not.